File: Selectionsort/selectionsort.js
//Run in Node.js
function swap(arr,a,b)
{
    let store = arr[a];
    arr[a] = arr[b];
    arr[b] = store;
}

function minPos(arr,start,end)
{
    let min = arr[start];
    let pos = start;
    for(let i=start; i<=end; i++)
    {    
        if(arr[i]<min)
        {
            min = arr[i];
            pos = i;
        }
    }
    return pos;
}

function selectionSort(arr)
{
    let length = arr.length-1;
    for(let i=0; i<length; i++)
    {
        let posMin = minPos(arr,i,length);
        swap(arr,posMin,i);
        console.log(arr);
    }
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

readline.question('Enter array (comma or space separated): ', (input) => 
{
    let arr = input.split(/[\s,]+/).map(Number); 

    console.log('Original array:', arr);

    selectionSort(arr);

    console.log('Sorted array:', arr);

    readline.close();
});

File: Selectionsort/selectionsort.py
def swap(arr,a,b):
    store = arr[a]
    arr[a] = arr[b]
    arr[b] = store

def min_pos(arr,start,end):
    min = arr[start]
    pos = start
    for i in range(start,end+1):
        if(arr[i]<min):
            min = arr[i]
            pos = i
    return pos

def selection_sort(arr):
    length = len(arr)-1
    for i in range(length):
        pos_min = min_pos(arr,i,length)
        swap(arr,pos_min,i)
        print(arr)
    
input_string = input("Enter array (comma or space separated): ")

arr = [int(x) for x in input_string.replace(',', ' ').split()]

print("Original array:", arr)

selection_sort(arr)

print("Sorted array:", arr)
